test/metabase/public_settings_test.clj

```diff
@@ -178,3 +179,24 @@
                  (public-settings/redirect-all-requests-to-https v)))
             (is (= false
                    (public-settings/redirect-all-requests-to-https)))))))))
+
+(deftest instance-creation-test
+  (let [original-value (db/select-one-field :value Setting :key "instance-creation")]
+    (try
+      (testing "Instance creation timestamp is set only once when setting is first fetched"
+        (db/delete! Setting {:key "instance-creation"})
+        (with-redefs [public-settings/first-user-creation (constantly nil)]
+          (let [first-value (public-settings/instance-creation)]
+            (Thread/sleep 10) ;; short sleep since java.time.Instant is not necessarily monotonic
+            (is (= first-value
+                   (public-settings/instance-creation))))))
+
+      (testing "If a user already exists, we should use the first user's creation timestamp"
+        (db/delete! Setting {:key "instance-creation"})
+        (mt/with-test-user :crowberto
+          (let [first-user-creation (:min (db/select-one ['User [:%min.date_joined :min]]))
```

Comment:
You don't need to bother to change this but you only need to quote stuff like `'User` to prevent circular references between model namespaces. e.g. if we want to refer to `Table` in `metabase.models.database` and `Database` in `metabase.models.table` ... one of those is going to have to quote the model so Toucan can do runtime resolution.

That stuff is going away in Toucan 2 since everything is keyword based so it's probably better to not use it where it's not needed. That way there's less stuff to change when we migrate over

Reply:
Ah yeah, I just copied this line over from the non-test code. I'll change it, no worries.